src/EchoToImageLUT.cpp

```diff
@@ -131,11 +131,14 @@ void EchoToImageLUT::drawImageFromEchoes(std::vector<uint8_t> const& world_echoe
     }
 }
 
-pair<vector<Point>::const_iterator, vector<Point>::const_iterator> EchoToImageLUT::
-    getPixels(int angle_idx, int sweep_idx) const
+pair<size_t, size_t> EchoToImageLUT::getPixels(int angle_idx, int sweep_idx) const
 {
-    int initial_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx];
-    int final_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1] - 1;
```

Comment:
The real fix is to remove this -1
```cpp
int final_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1];
```

if `_data_index[angle_idx * m_sweep_size + sweep_idx + 1] == 0` it was becaming -1